php-frontend/src/test/java/org/sonar/php/regex/PhpRegexSourceTest.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * SonarQube PHP Plugin
+ * Copyright (C) 2010-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.php.regex;
+
+import java.util.List;
+import org.junit.Test;
+import org.sonarsource.analyzer.commons.regex.RegexParseResult;
+import org.sonarsource.analyzer.commons.regex.RegexParser;
+import org.sonarsource.analyzer.commons.regex.RegexSource;
+import org.sonarsource.analyzer.commons.regex.ast.CharacterTree;
+import org.sonarsource.analyzer.commons.regex.ast.FlagSet;
+import org.sonarsource.analyzer.commons.regex.ast.RegexTree;
+import org.sonarsource.analyzer.commons.regex.ast.SequenceTree;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
+import static org.sonar.php.regex.RegexParserTestUtils.assertKind;
+import static org.sonar.php.regex.RegexParserTestUtils.assertSuccessfulParse;
+import static org.sonar.php.regex.RegexParserTestUtils.makeSource;
+
+public class PhpRegexSourceTest {
+
+  @Test
+  // TODO: Extend test with exact syntax error location check
+  public void invalid_regex() {
+    RegexSource source = makeSource("'/+/'");
+    RegexParseResult result = new RegexParser(source, new FlagSet()).parse();
+
+    assertThat(result.getSyntaxErrors()).isNotEmpty();
+  }
+
+  @Test
+  public void test_missing_delimiters() {
+    Exception exception = assertThrows(IllegalArgumentException.class, () -> makeSource("'a\nb'"));
+    assertEquals(exception.getMessage(), "Regular expression does not contain delimiters");
```

Comment:
You may also use `org.assertj.core.api.Assertions.assertThatThrownBy`, see [an example](https://github.com/SonarSource/sonar-php/blob/master/sonar-php-plugin/src/test/java/org/sonar/plugins/php/PHPSensorTest.java#L292-L295).